### ISSUE TITLE
Issues with postcss-loader

### ISSUE THREAD

**Issue by erikaheidi** [tailwindlabs/tailwindcss-typography#147]
Hello! :wave: 

I am having some issues in getting this plugin installed. I don't have much experience with NPM, but tried to change the version of `postcss-loader` after finding some people mention problems on Stackoverflow, but it didn't solve my problem. 

When I run `npm run dev` with version `^4.2.0` of `postcss-loader`, I get the following errors:

```

 ERROR  Failed to compile with 2 errors                                                        9:29:29 AM

 error  in ./resources/css/app.css

Module build failed (from ./node_modules/css-loader/index.js):
ModuleBuildError: Module build failed (from ./node_modules/postcss-loader/dist/cjs.js):
ValidationError: Invalid options object. PostCSS Loader has been initialized using an options object that does not match the API schema.
 - options has an unknown property 'plugins'. These properties are valid:
   object { postcssOptions?, execute?, sourceMap?, implementation? }
    at validate (/var/www/node_modules/postcss-loader/node_modules/schema-utils/dist/validate.js:104:11)
    at Object.loader (/var/www/node_modules/postcss-loader/dist/index.js:43:29)
    at /var/www/node_modules/webpack/lib/NormalModule.js:316:20
    at /var/www/node_modules/loader-runner/lib/LoaderRunner.js:367:11
    at /var/www/node_modules/loader-runner/lib/LoaderRunner.js:233:18
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

 error  in ./resources/css/app.css

Module build failed (from ./node_modules/postcss-loader/dist/cjs.js):
ValidationError: Invalid options object. PostCSS Loader has been initialized using an options object that does not match the API schema.
 - options has an unknown property 'plugins'. These properties are valid:
   object { postcssOptions?, execute?, sourceMap?, implementation? }
    at validate (/var/www/node_modules/postcss-loader/node_modules/schema-utils/dist/validate.js:104:11)
    at Object.loader (/var/www/node_modules/postcss-loader/dist/index.js:43:29)

 @ ./resources/css/app.css 2:14-141
```

If I use the newest `postcss-loader` version, then I get different errors:

```
 ERROR  Failed to compile with 2 errors                                                        9:38:14 AM

 error  in ./resources/css/app.css

Module build failed (from ./node_modules/css-loader/index.js):
ModuleBuildError: Module build failed (from ./node_modules/postcss-loader/dist/cjs.js):
TypeError: this.getOptions is not a function
    at Object.loader (/var/www/node_modules/postcss-loader/dist/index.js:38:24)
    at /var/www/node_modules/webpack/lib/NormalModule.js:316:20
    at /var/www/node_modules/loader-runner/lib/LoaderRunner.js:367:11
    at /var/www/node_modules/loader-runner/lib/LoaderRunner.js:233:18
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

 error  in ./resources/css/app.css

Module build failed (from ./node_modules/postcss-loader/dist/cjs.js):
TypeError: this.getOptions is not a function
    at Object.loader (/var/www/node_modules/postcss-loader/dist/index.js:38:24)

 @ ./resources/css/app.css 2:14-141
```

I also tried running `composer update` and `npm update` but nothing seemed to work. Everything was working well and I just wanted to make my blog posts look nice :sweat_smile: does anyone have any leads on what's the issue? 

Thank you in advance!

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

One thing I notice from the first error is `options has an unknown property 'plugins'. These properties are valid:`. Are you defining this plugin in your `postcss.config.js` file instead of your `tailwind.config.js` file?

Could you perhaps shared those config files, or provide a (small) reproduction repository that showcases this error?

**Comment by RobinMalfait** [comment]
Hey!

I created an example repo with all the steps I took to get to the result, here is the repo: https://github.com/RobinMalfait/laravel-blog-example/commits/main
The final result looks like this: 
<img width="1501" alt="image" src="https://user-images.githubusercontent.com/1834413/110950133-bead2d80-8343-11eb-9efe-6caebd2c05fd.png">

A few things to notice here:

1. I am on node version `v15.11.0`, but anything above `v12.13.0` should suffice. You can verify this using `node -v`.
2. Postcss is on version `^8.2.8`
3. laravel-mix is on version `^6.0.6`
4. I followed the steps here: https://tailwindcss.com/docs/guides/laravel and committed every step.
5. After I added the markdown and the typography plugin, I re-ran `npm run prod`

Could you perhaps clone that repo and see if that works on your machine? (`php artisan serve` should do he trick)

Sorry for the inconvenience!

**Comment by erikaheidi** [comment]
Hey @RobinMalfait thank you so much for your detailed reply, it helped me figure out the issue! I was using an older version of `laravel-mix` (5.0.2), and after updating it I just needed to remove  the `--hide-modules` option from the `package.json` file and I was able to run `npm run dev` without issues. It all works now!

It seems like I was using an older version of the default one that comes with Laravel, I think! This project went through some updates, and I only started using Tailwind recently. Anyways, it looks great now, thank you so much for all your help!
